### ISSUE TITLE
Random Sampling feature on different point cloud density 

### ISSUE THREAD

**Issue by abhigoku10** [QingyongHu/RandLA-Net#62]
@QingyongHu i have few queries w r t sampling feature 
Q1. As mentioned in the paper you have perfomed random sampling sinc the point cloud density was 10^5 , should we make change in the source code if we have the point cloud density to be 10^3/ 10^4
Q2 can we disable the random sampling few if we have less points in the point cloud is there any option available

**Comment by QingyongHu** [comment]
Hi @abhigoku10 , sorry for the late response. I was over-busy these days.
Q1: Of course you should change, downsampling essentially breaks the spatial structure of point clouds, especially when the density is low
Q2: It is definitely ok if you do not use random sampling. This paper aim at processing large-scale point clouds, so we choose random sampling to balance efficiency and performance. For small point clouds, farthest point sampling and other learning-based sampling are worth to try.
Q3: To alleviate the problem of class imbalance, we use the weighted cross-entropy as the loss function. The weight of each class is determined by its inverse frequency in the training split.

**Comment by abhigoku10** [comment]
@QingyongHu  thanks for your response , i have follow up queries 

Q1. IN the paper you have mentioned that there are 4 encoding and decoding layers to progressively reduce the size of the large point cloud , is 4 layers an experimented set value ? can we modify the 4 layers to 3/2 layers for small scale point clouds ??
Q2.last, three fully-connected layers and a dropout layer are used to predict the semantic label
of each point---instead of using fc we could use 1x1 conv right is any specific to use 3 fc 
Q3. Have you performed any model interpertabliity feature to understand hwo the features are getting learned more for cars and less for bicycle and all 

Thanks in advance and your formidable work
